crates/uv/src/commands/python/pin.rs

```diff
@@ -68,16 +68,27 @@ pub(crate) async fn pin(
         request.to_canonical_string()
     };
 
-    debug!("Using pin `{}`", output);
-    let version_file = PathBuf::from(PYTHON_VERSION_FILENAME);
-    let exists = version_file.exists();
+    let existing = request_from_version_file().await.ok().flatten();
+    write_version_file(&output).await?;
 
-    debug!("Writing pin to {}", version_file.user_display());
-    fs_err::write(&version_file, format!("{output}\n"))?;
-    if exists {
-        writeln!(printer.stdout(), "Replaced existing pin with `{output}`")?;
+    if let Some(existing) = existing
+        .map(|existing| existing.to_canonical_string())
+        .filter(|existing| existing != &output)
+    {
+        writeln!(
+            printer.stdout(),
+            "Updated `{}` from `{}` to `{}`",
```

Comment:
Should we use a right arrow symbol, e.g., →?

Reply:
Done, I used the same arrow as in `uv lock --upgrade`